<img style="float: left; margin: 30px 15px 15px 15px;" src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFzQj91sOlkeDFkg5HDbjtR4QJYmLXkfMNig&usqp=CAU" width="400" height="500" /> 
    
    
### <font color='navy'> Simulación matemática. 

**Nombre:** TALPA DE JESUS ORNELAS GÜITRON.

**Fecha:** 1 de noviembre del 2023.

**Expediente** : 745134.
    
**Profesor:** Oscar David Jaramillo Zuluaga.
    
**Link Github**: 

## TAREA 8 Modelo de Ahorro

## Código Estudiante 1

In [96]:
# Librería de análisis de datos
import pandas as pd
# Librería para manipulación de fechas
from datetime import date
import numpy as np
import numpy_financial as npf

## Datos de Banco Azteca

### Plazo Azteca
> - Es una inversión a plazo fijo en moneda nacional de cobertura nacional que te paga intereses al vencimiento.
> - Comienza a invertir con un depósito inicial de $\$15,000.00$ pesos
>
> Considerando que los plazos y las tazas de intereses cambian de acuerdo al tiempo de inversión del plazo, se elegió considerado una inversión azteca que se renueva cada 91 dias, en la cual al finalizar el tiempo se pagan los intereses y al mismo tiempo se vuelve a hacer la inversión si así lo decides, en este tiempo es cuando nosotros realizaremos un abono para seguir generando ganancias en lo que estemos acumulando cada 3 meses aproximadamente.

https://www.bancoazteca.com.mx/productos/inversion/inversion-azteca-plazo.html

### Datos

> - La inversion elegida al plazo de 91 dias tiene GAT Nominal (Ganancia Anual Total Nominal) **2.78% = tasa 0.0278**
> - El plazo total de la inversión para generar el dinero a meta es de **3 años**, puesto que es el aproximado para terminar mi carrera en la universidad
> - Al ser cada **91 dias**, quiere decir que en el año se daran **4 abonos**
> - El monto de ahorro inicial que se tomó en cuenta es el mínimo en Banco Azteca $\$15,000.00$

In [97]:
# Datos para la cuenta de ahorro
tasa_interes = 0.0278
years = 3
Abono_year = 4
inicial = 15000
meta = 85000
fecha_inicio = (date(2023,10,30))

In [98]:
# Cálculo de abonos mensuales iguales para llegar a la meta de ahorro en el tiempo deseado
Abono = -npf.pmt(tasa_interes/Abono_year,years*Abono_year,(meta-inicial))
Abono

6100.199859247117

Con esto obtenemos que el abono necesario a cada vencimiento (91 días) es de $\$6,100.20$ aproximadamente, considerando el plazo total, los abonos por año y la tasa de rendimiento que nos ofrece la institución bancaria.

In [99]:
# Rango de fechas en la tabla
rng = pd.date_range(fecha_inicio, periods=years*Abono_year, freq='91D')
rng.name = 'Fecha del abono'

# Columnas de abonos y extracto de cuenta en las fechas respectivas
df = pd.DataFrame(index=rng, columns=['Abono', 'Balance'], dtype='float')

# Para poner el indice con los periodos y no con las fechas
df.reset_index(inplace=True)
df.index += 1
df.index.name = 'Periodo'

# Agregar una nueva columna que represente la fecha del final de periodo
df['Fin del periodo'] = rng + pd.DateOffset(days=91)


In [100]:
npf.ppmt(tasa_interes/Abono_year, 1, years * Abono_year, Abono)

-489.20987273192617

In [101]:
# Los abonos son iguales
df['Abono'] = Abono
# Vector indizador de los periodos de los abonos
index_vector = np.arange(1,len(rng)+1)
# Extracto de la cuenta mes a mes
df["Balance"] = inicial * (1 + tasa_interes/Abono_year)**index_vector + \
                Abono * (((1 + tasa_interes/Abono_year)**index_vector - 1)/(tasa_interes/Abono_year))


In [102]:
# Redondear con dos cifras decimales y mostrar DataFrame
df = df.round(2)
df

Fecha del abono   Abono   Balance Fin del periodo
Periodo                                                  
1            2023-10-30  6100.2  21204.45      2024-01-29
2            2024-01-29  6100.2  27452.02      2024-04-29
3            2024-04-29  6100.2  33743.01      2024-07-29
4            2024-07-29  6100.2  40077.73      2024-10-28
5            2024-10-28  6100.2  46456.47      2025-01-27
6            2025-01-27  6100.2  52879.54      2025-04-28
7            2025-04-28  6100.2  59347.25      2025-07-28
8            2025-07-28  6100.2  65859.91      2025-10-27
9            2025-10-27  6100.2  72417.84      2026-01-26
10           2026-01-26  6100.2  79021.34      2026-04-27
11           2026-04-27  6100.2  85670.74      2026-07-27
12           2026-07-27  6100.2  92366.35      2026-10-26

### Resultados

Finalmente en la tabla se muestran los rendimientos después de cada periodo correspondiente, como podemos observar, si comenzamos a realizar el ahorro a partir del 30 de octubre de 2023 con el capital inicial de 15,000 más el abono correspondiente de 6,100 tardaríamos 11 periodos en obtener nuestro objetivo de ahorro, lo que nos dice que en 11 periodos obtendríamos $\$85,670.74$ , lo que significa poco más de la meta, considerando que al final del periodo es cuando se pagan los intereses, se ha agregado una columna mas que representa el fin del mismo es decir, la fecha en que se obtiene la inversión con sus respectivas ganancias, que será igual a la fecha en la se abona e inicia el siguiente periodo.

Ya con todo esto, considerando el punto de la fecha del final del periodo podemos decir que duraríamos aproximadamente 2 años y 9 meses en obtener nuestro ahorro meta de $\$80,000.00$ iniciando el 30 de octubre de 2023 y finalizando el 26 de julio de 2026, considerando que nuestro capital inicial es de $\$15,000.00$ más el abono correspondiente de $\$6,000.00$ 


In [ ]:
....